hack/update-helmgen.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+# Copyright 2021 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ -n "${BUILD_WORKSPACE_DIRECTORY:-}" ]]; then # Running inside bazel
+  IN_DIR="$BUILD_WORKSPACE_DIRECTORY/deploy/charts/cert-manager"
+  OUT_DIR="$BUILD_WORKSPACE_DIRECTORY/deploy/charts/cert-manager"
+  echo $BUILD_WORKSPACE_DIRECTORY
+fi
+
+if [[ -n "${IN_DIR:-}" && -n "${OUT_DIR:-}" ]]; then # Running as part of a bazel genrule
+  echo "Updating templated helm README.md & Chart.yaml..." >&2
+elif ! command -v bazel &>/dev/null; then
+  echo "Install bazel at https://bazel.build" >&2
+  exit 1
+else
+  (
+    set -o xtrace
+    bazel run //hack:update-helmgen
+  )
+  exit 0
+fi
+
+if [[ "$1" == "default" ]]; then 
+  VERSION="v0.0.1"
+  IS_PRERELEASE="true"
+else
+  version=$PWD/$1
+  VERSION=$(cat "$version")
+  IS_PRERELEASE=$(grep -q '^v[0-9]\\+.[0-9]\\+.[0-9]\\+$$' "$version" && echo "false" || echo "true")
+fi
+
+jq=$PWD/$2
+chart_doc_gen=$PWD/$3
+renderizer=$PWD/$4
+
+shift 4
+
+doc=$(mktemp -t doc.XXXX.yaml)
+
+c="."
+c=$c' | .Repository.Name="jetstack"'
+c=$c' | .Repository.URL="https://charts.jetstack.io"'
+c=$c' | .Chart.Name="cert-manager"'
+c=$c' | .Chart.Version=$version'
+c=$c' | .Chart.IsPrerelease=$is_prerelease'
+c=$c' | .Release.Name="my-cert-manager"'
+c=$c' | .Release.Namespace="cert-manager"'
+
+echo "{}" | "$jq" \
+  --arg version $VERSION \
+  --arg is_prerelease $IS_PRERELEASE \
+  "$c" > "$doc"
+
+cat "$doc"
```

Comment:
Are you using `jq` to build a json config file for renderizer. That's clever.

But does renderizer accept a yamle config file, and if so, can't the yaml be supplied using a HEREDOC which I think would be easier to read and would avoid having to depend on `jq`.

Reply:
I'll try to remove the `jq` dependency.